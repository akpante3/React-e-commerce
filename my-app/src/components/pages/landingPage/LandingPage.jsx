import React, { Component } from 'react';
import { connect } from 'react-redux';
import { CSSTransition } from 'react-transition-group';
import { bindActionCreators } from 'redux';
import ModalImage from '../../ui/modals/modalImage/ModalImage';
import './LandingPage.scss';
import AppButton from '../../ui/buttons/AppButton/AppButton';
import ProductFilterCard from '../../ui/cards/productFilterCard/ProductFilterCard';
import ProductCard from '../../ui/cards/productCard/ProductCard';
import Pagination from '../../ui/pagination/Pagination';
import Spinner from '../../ui/Spinner/Spinner';


import {
  getAllProducts,
  getProductsByCatergory,
  getProductsByDepartment,
} from '../../../store/actions/getProducts';


class LandingPage extends Component {
  state = {

  };

  componentDidMount() {
    this.props.actions.getAllProducts(1);
  }

  paginateProduct = (page) => {
    if (this.props.products.productType === 'all-products') {
      this.props.actions.getAllProducts(page);
    }
    if (this.props.products.productType === 'department-products') {
      this.props.actions.getProductsByDepartment(page, this.props.products.id);
    }
    if (this.props.products.productType === 'category-products') {
      this.props.actions.getProductsByCatergory(page, this.props.products.id);
    }
  }

  render() {
    const { products } = this.props;
    return (products.products ? (
      <CSSTransition
        in
        appear
        timeout={3000}
        classNames="fade"
      >
        <div className="landing-page">
          <ModalImage
            image='https://raw.githubusercontent.com/zandoan/turing-uiux/master/Zeplin/Images/Images-modal4.png'
            details={(
              <div>
                <span>Mens wear</span>
                <div><AppButton title='shop' /></div>
              </div>
            )}
          />
          <div className="landing-page-content mt-5">
            <div className="landing-filter-card">
              <ProductFilterCard
                count={products.count}
              />
            </div>
            {
            products.products
              ? products.products.map(product => (
                <div
                  className="landing-page-product"
                  key={product.product_id}
                >
                  <ProductCard
                    product={product}
                  />
                </div>
              )) : null
          }
          </div>
          <div className="pagination text-center mt-5 mb-5">
            <Pagination
              count={products.count}
              paginateProduct={this.paginateProduct}
            />
          </div>
          <ModalImage
            image='https://www.playgroundshop.com/upload/comunicazioni_home/190402_GALLERY_DONNA_EN.gif'
          />
        </div>
      </CSSTransition>
    ) : <Spinner />
    );
  }
}


const mapStateToProps = state => ({
  categories: state.categories.categories,
  products: state.products,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    { getAllProducts, getProductsByCatergory, getProductsByDepartment },
    dispatch,
  ),
});

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage);
