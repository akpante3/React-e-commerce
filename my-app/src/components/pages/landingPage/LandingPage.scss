.landing-page {
  width: 80%;
  margin: 0 auto;
  .landing-page-content {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-template-rows: repeat(2, 1fr);
    grid-gap: 0.8em;
    .landing-filter-card {
      grid-row-end: span 2;
      margin: 0 auto;
    }
    .landing-page-product {
      grid-area: "product";
      margin: 0 auto;
    }
  }
  .pagination {
    max-width: 100%;
    justify-content: center;
    align-self: center
  }
}

.fade-appear {
  opacity: 0;
  z-index: 1;
}

.fade-appear.fade-appear-active {
  opacity: 1;
  transition: opacity 1500ms linear 1500ms;
}
@media only screen and (max-width: 1100px) {
  .landing-page {
    width: 90%;
    .landing-page-content {
      grid-template-columns: 1fr 1fr 1fr;
      grid-gap: 0.7em;
    }
  }
}

@media only screen and (max-width: 900px) {
  .landing-page {
    width: 90%;
    .landing-page-content {
      grid-template-columns: 1fr 1fr;
    }
  }
}
@media only screen and (max-width: 450px) {
  .landing-page {
    width: 90%;
    .landing-page-content {
      grid-template-columns: 1fr;
    }
  }
}
