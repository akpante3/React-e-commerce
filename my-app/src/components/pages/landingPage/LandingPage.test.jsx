import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import LandingPage from './LandingPage';

const initialState = {
  categories: { categories: [] },
  products: { products: [] },
};
const mockStore = configureStore();
let wrapper;
let store;

describe('test for LandingPage', () => {
  it('test for LandingPage', () => {
    store = mockStore(initialState);
    wrapper = shallow(<LandingPage store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
