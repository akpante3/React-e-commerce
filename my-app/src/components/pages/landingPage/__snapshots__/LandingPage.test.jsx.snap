// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for LandingPage test for LandingPage 1`] = `
<ContextProvider
  value={null}
>
  <LandingPage
    actions={
      Object {
        "getAllProducts": [Function],
        "getProductsByCatergory": [Function],
        "getProductsByDepartment": [Function],
      }
    }
    categories={Array []}
    products={
      Object {
        "products": Array [],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
